### PR TITLE
Task six

### PR DESCRIPTION
**Pull request by barty008**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -303,3 +303,82 @@ describe("/api/articles", () => {
...
+        ).length
+        expect(receivedComments.length).toBe(expectedNumberOfComments)
+        receivedComments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number),
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: you know the article id

File: __tests__/app.test.js [hunk 2]
```diff
@@ -303,3 +303,82 @@ describe("/api/articles", () => {
...
+      .expect(200)
+      .then(({ body }) => {
+        const { comments: receivedComments } = body
+        expect(receivedComments).toBeSortedBy("created_at", {
+          descending: true,
+        })
+        done()
+      })
+      .catch(done)
+  })
```
Comment (Hard-Luck) [inline __tests__/app.test.js:384]: A couple more tests are required
- what if the article id is not found?
- what if the article id is invalid?

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Good start, just needs some more tests

File: __tests__/app.test.js [hunk 3]
```diff
@@ -303,3 +303,102 @@ describe("/api/articles", () => {
...
+        const expectedNumberOfComments = modifiedCommentData.filter(
+          (comment) => {
+            return comment.article_id === 1
+          }
+        ).length
+        expect(receivedComments.length).toBe(expectedNumberOfComments)
+        done()
+      })
+      .catch(done)
+  })
```
Comment (Harrison365) [inline __tests__/app.test.js:328]: Not sure why we have a done() here. We also don't need catch blocks in our tests. You can remove these :)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -303,3 +303,102 @@ describe("/api/articles", () => {
...
+      .expect(200)
+      .then(({ body }) => {
+        const { comments: receivedComments } = body
+        console.log(receivedComments, "<----")
+        expect(receivedComments).toEqual([])
+        done()
+      })
+      .catch(done)
+  })
+
```
Comment (Harrison365) [inline __tests__/app.test.js:342]: Nice test 👍 
(again lose the done and the catch)

File: __tests__/app.test.js (hunk 1 above)
Comment (Harrison365) [reply __tests__/app.test.js]: ^^^ still need to address this please :)

File: __tests__/app.test.js [hunk 5]
```diff
@@ -303,3 +303,102 @@ describe("/api/articles", () => {
...
+  test("GET: 400 for invalid article_id type should return 'Bad Request' message", () => {
+    return request(app)
+      .get(`/api/articles/unknown/comments`)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.status).toBe(400)
+        expect(body.msg).toBe("Bad Request")
+      })
+  })
```
Comment (Harrison365) [inline __tests__/app.test.js:404]: Nice sad-path tests. Well done :)

File: controller/controller.js [hunk 6]
```diff
@@ -51,7 +53,25 @@ exports.getAllArticles = (req, res, next) => {
...
+  //1 need to retrieve comments for the articleid
+  // 2 need to check if the artile exists, create a funciton for this = needs to be in model
+  const allPromises = [
+    selectComments(article_id),
+    checkArticleExists(article_id),
+  ]
+  Promise.all(allPromises)
+    .then(([comments]) => {
+      // console.log(comments, "<-------")
+      res.status(200).send({ comments })
```
Comment (Harrison365) [inline controller/controller.js:74]: Nice use of Promise.all() 👍

File: app.js [hunk 7]
```diff
@@ -3,27 +3,50 @@ const {
   getTopics,
   getEndPoints,
   getArticleById,
-  getAllArticles
+  getAllArticles,
+  getComments,
 } = require("./controller/controller")
+const error404 = (req, res, next) => {
+  res.status(404).send({ status: 404, msg: "Not Found" })
+}
```
Comment (Harrison365) [inline app.js:15]: We shouldn't be hard coding this. It needs to take the custom error message and return that :) 
Please let us know if you aren't sure what I mean by this 👍

**Pull request merged**